era_5g_interface/h264_decoder.py

```diff
@@ -107,20 +106,29 @@ def decode_packet_data(self, packet_data: bytes, format: str = "bgr24") -> np.nd
             Video frame / image.
         """
 
-        packet = Packet(packet_data)
-        # TODO: only for testing purpose
-        # logger.info(f"Decoding packet: {packet}")
-
-        # Multiple frames? - This should not happen because on the encoders side one frame is always encoded and sent
-        frame: VideoFrame
-        for frame in self._decoder.decode(packet):
-            # TODO: only for testing purpose
-            # logger.info(f"Frame {frame} with id {frame.index} decoded from packet: {packet}")
-            # logger.info(f"frame.pts: {frame.pts}, frame.dts: {frame.dts}, frame.index: {frame.index}, "
-            #            f"frame.key_frame: {frame.key_frame}, frame.is_corrupt: {frame.is_corrupt}, "
-            #            f"frame.time: {frame.time}")
-            # frame.to_image().save('output/frame-%04d.jpg' % frame.index)
-
-            self._last_frame_is_keyframe = frame.key_frame
-        frame_ndarray: np.ndarray = frame.to_ndarray(format=format)
-        return frame_ndarray
+        try:
+            packet = Packet(packet_data)
+
+            # Multiple frames? - This should not happen because on the encoders side one frame is always encoded and sent
+            frame: VideoFrame
+            for frame in self._decoder.decode(packet):
+                logger.debug(f"Frame {frame} with id {frame.index} decoded from packet: {packet}")
+                logger.debug(
+                    f"frame.pts: {frame.pts}, "
+                    f"frame.dts: {frame.dts}, "
+                    f"frame.index: {frame.index}, "
+                    f"frame.key_frame: {frame.key_frame}, "
+                    f"frame.is_corrupt: {frame.is_corrupt}, "
+                    f"frame.time: {frame.time}"
+                )
+                # TODO: only for testing purpose
+                # frame.to_image().save('output/frame-%04d.jpg' % frame.index)
+
+                self._last_frame_is_keyframe = frame.key_frame
+                frame_ndarray: np.ndarray = frame.to_ndarray(format=format)
+                return frame_ndarray
+            # Sometimes, with dropping some TCP packets, no frame is decoded from av.Packet while no FFmpegError is
+            # raised. This is resolved by throwing this exception.
+            raise H264DecoderError(f"No frame decoded from packet {packet}")
+        except Exception as e:
```

Comment:
What about something like this...

```python
try:
    frames = self._decoder.decode(packet)  # weird stuff can happen here
except Exception:
    ...

# this is pretty safe
for frame in frames:
    ...
```

Now, `except Exception` will catch for instance `AttributeError` if you try to log something that is now available (typo in property name).

Reply:
I don't think it will work that way because it is an iterator. I don't have many other ideas to make it better...

Reply:
I originally wanted to generate a new FFmpegError, but it is bound to FFmepg.

Reply:
Ok, iterator, but you anyway just want only the first frame, right? And there should be just one. Then, you can convert it to a list and take the first element:

```python
try:
    frame = list(self._decoder.decode(packet))[0]
except IndexError:
    print("No frame!")
except Exception:
    print("Big fail")

# some operation with a frame
```

Reply:
Did you see the new code?  I solved it a little differently.  I'll still try converting to list, but I'm not sure it will work.  I think I've tried something similar before.